op-node/p2p/host.go

```diff
@@ -249,6 +255,14 @@ func (conf *Config) Host(log log.Logger, reporter metrics.Reporter, metrics Host
 		staticPeers: staticPeers,
 		quitC:       make(chan struct{}),
 	}
+
+	if conf.EnablePingService {
+		out.pinging = NewPingService(log,
+			func(ctx context.Context, peerID peer.ID) <-chan ping.Result {
+				return ping.Ping(ctx, h, peerID)
+			}, h.Network().Peers, clock.SystemClock)
+	}
```

Comment:
what happens for peers who don't enable this or haven't upgraded? I assume they just get warnings about unsupported messages?

Reply:
Server side: serving pings was already enabled, it's on by default, and we don't support disabling it. If it's somehow disabled though, the libp2p stack will fail to negotiate the ping-protocol, and ping attempts will fail with an error.
Client side: requesting pings is optional, it fills latency data, but not doing the pings is fine.